aiosmtpd/smtp.py

```diff
@@ -182,18 +182,26 @@ def _handle_client(self):
             line = yield from self._reader.readline()
             try:
                 # XXX this rstrip may not completely preserve old behavior.
-                line = line.decode('utf-8').rstrip('\r\n')
+                line = line.rstrip(b'\r\n')
                 log.info('Data: %r', line)
                 if not line:
                     yield from self.push('500 Error: bad syntax')
                     continue
-                i = line.find(' ')
+                i = line.find(b' ')
+                # keep data not decoded, decode only command name part as
+                # it can be only in ASCII encoding. Let commands decide
+                # what encoding to expect
```

Comment:
Hi! My first implementation here was to keep them bytes, thats why comments say that. I missed to update them. But that makes code very ugly, as every command it do same magic on start. Really there is no use case on protocol level to worry about paramenters encoding problems. All such cases are on user side and must be controlled under handler implementations.